jwala-webservices/src/main/java/com/cerner/jwala/ws/rest/v1/service/group/impl/GroupServiceRestImpl.java

```diff
@@ -337,56 +337,89 @@ public Response generateAndDeployGroupWebServersFile(final String groupName, fin
 
-                    }
-                });
-                futureMap.put(name, responseFuture);
-            }
             checkResponsesForErrorStatus(futureMap);
         } else {
             LOGGER.info("No web servers in group {}", groupName);
         }
-        return ResponseBuilder.ok(httpdTemplateContent);
+        return ResponseBuilder.ok(resourceTemplateContent);
     }
 
-    protected void checkResponsesForErrorStatus(Map<String, Future<Response>> futureMap) {
+    private Map<String, Future<Response>> executeWebServerResourceDeploy(final String groupName, final String resourceFileName, final AuthenticatedUser aUser, final String httpdTemplateContent, final String resourceMetaData, Set<WebServer> webServers) {
+        final Map<String, Future<Response>> futureMap = new HashMap<>();
+        final Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+        for (final WebServer webserver : webServers) {
+            final String name = webserver.getName();
+            Future<Response> responseFuture = executorService.submit(new Callable<Response>() {
+                @Override
+                public Response call() throws Exception {
+                    SecurityContextHolder.getContext().setAuthentication(auth);
+                    webServerServiceRest.updateResourceTemplate(name, resourceFileName, httpdTemplateContent);
+                    ResourceIdentifier resourceId = new ResourceIdentifier.Builder()
+                            .setGroupName(groupName)
+                            .setWebServerName(name)
+                            .setResourceName(resourceFileName)
+                            .build();
+                    resourceService.updateResourceMetaData(resourceId, resourceFileName, resourceMetaData);
+                    return webServerServiceRest.generateAndDeployConfig(name, resourceFileName, aUser);
+
+                }
+            });
+            futureMap.put(name, responseFuture);
+        }
+        return futureMap;
+    }
+
+    private void checkWebServerStateBeforeDeploy(String resourceFileName, Group groupWithWebServers, Set<WebServer> webServers) {
+        List<String> startedWebServers = new ArrayList<>();
+        for (WebServer webServer : webServers) {
+            ResourceIdentifier resourceIdentifier = new ResourceIdentifier.Builder()
+                    .setWebServerName(webServer.getName())
+                    .setResourceName(resourceFileName)
+                    .setGroupName(groupWithWebServers.getName())
+                    .build();
+            String metaDataStr = resourceService.getResourceContent(resourceIdentifier).getMetaData();
+            try {
+                ResourceTemplateMetaData metaData = resourceService.getTokenizedMetaData(resourceFileName, webServer, metaDataStr);
+                if (webServerService.isStarted(webServer)) {
+                    if (metaData.isHotDeploy()) {
```

Comment:
Can we change this to

    if (webServerService.isStarted && !metaData.isHotDeploy) {
        ...
        continue;
    }
    LOGGER.info(...);